### ISSUE TITLE
[Bug]: Screen readers are announcing the selected date as current date in DatePicker

### ISSUE THREAD

**Issue by Aakash3161** [microsoft/fluentui#26750]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (8) x64 Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz
    Memory: 18.64 GB / 31.59 GB
  Browsers:
    Edge: Spartan (44.22621.1105.0), Chromium (108.0.1462.76)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://developer.microsoft.com/en-us/fluentui#/controls/web/datepicker

## Actual Behavior
Screen readers are announcing the selected date as current date in [DataPicker](https://developer.microsoft.com/en-us/fluentui#/controls/web/datepicker), because aria-current="date" attribute is assigned to 'Selected' date. 

## Expected Behavior
aria-current="date" attribute should represent the current date ( today's date ) in a calendar. aria-selected should be used to represent the 'Selected' date in a calendar.

### Requested priority

Normal

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.
